tensorflow/compiler/mlir/xla/transforms/hlo_legalize_to_lhlo.cc

```diff
@@ -194,11 +194,50 @@ class HloToLhloTensorStoreOpConverter : public ConversionPattern {
   PatternMatchResult matchAndRewrite(
       Operation* op, ArrayRef<Value> operands,
       ConversionPatternRewriter& rewriter) const final {
-    rewriter.eraseOp(op);
+    rewriter.replaceOpWithNewOp<xla_lhlo::CopyOp>(
+        op, llvm::None, operands.front(), operands.back());
     return matchSuccess();
   }
 };
 
+/// Removes an allocated buffer without any users.
+void RemoveRedundantAllocDealloc(ModuleOp module) {
```

Comment:
do we need this pass? can it be combined with RemoveRedundantCopyOp pass?

Reply:
The redundant deallocOp cannot be erased in the first pass since it causes a crash in the module pass. I created an eraseList to keep track of operations need to be removed and erased them after the pass.